sqlmesh/core/model/meta.py

```diff
@@ -41,9 +42,9 @@ class IntervalUnit(str, Enum):
 class ModelMeta(PydanticModel):
     """Metadata for models which can be defined in SQL."""
 
+    dialect: str = ""
     name: str
     kind: ModelKind = ModelKind(name=ModelKindName.VIEW)
-    dialect: str = ""
```

Comment:
The `dialect` was moved before `name` in order to reference it in `name`'s validator (see below).